docs/enhancements/soft-install-timeout.md

```diff
@@ -0,0 +1,172 @@
+| Wait for finalizing        | installation (cluster)| controller | Assisted controller | hard coded |                       10h | Don't perform post install + don't send complete installation |
+| Wait for cluster operators | installation (cluster)| controller | Assisted controller | hard coded |                       10h |  Don't perform rest of controller operations | Only CVO and console |
+| Add router CA              | installation (cluster)| controller | Assisted controller | hard coded |                       70m | Don't perform rest of controller operations |
+| Wait for OLM operators     | installation (cluster)| controller | Assisted controller || calculated from operators | | Don't perform rest of controller operations |
+| Apply manifests            | installation (cluster)| controller | Assisted controller | hard coded |                       10m |  Don't perform rest of controller operations |
+| Wait for OLM operators CSV | installation (cluster)| controller | Assisted controller | | calculated from operators | Don't perform rest of controller operations |
+| Send complete installation | installation (cluster)| controller | Assisted controller | hard coded | 30m |  Don't perform rest of controller operations |
+
+
+There are 2 flows for completing installation.  One is managed by controller, and one by assisted service.
+
+In the flow managed by the assisted service, the following steps must be completed:
+
+- kubeconfig must be uploaded
+- cluster operators are successful
+- Monitored operators are either successful or failed
+
+In the flow managed by the controller all steps must be completed (in the table above).  The last step is a 
+a notification to complete installation.
+
+### Suggested changes
+- Only assisted service should notify installation completion
+- All steps by controller will be known as stages in the assisted service.  This will enable the service to manage 
+them in a similar way that host stages are managed.
+- Controller will not stop activity due to timeout.
+- Assisted service should be able to terminate the controller.
+- All existing timeouts except installation timeout and rebooting timeout should be treated as soft timeouts (i.e notification only)
+- In case of some optional operation (i.e managed operators), the user will be able to decide to skip it and set the cluster as completed.
+
+### Open Questions
+
+- Do we also soften the 24h `INSTALLATION_TIMEOUT` timeout?
+
+- Should we collect must gather logs when the installation timeout expires, even
+  if we don't mark it as failed?
+- Should any of these changes be optional ?
+
+### UI Impact
+
+The UI will need to explicitly show to the user that the cluster installation
+is taking longer than expected, and give suggestions on how to proceed. For
+example, we could present a warning message with this text:
+
+> Cluster installation is taking too long
+>
+> It should have completed in approximately 45 minutes, but it took 23 hours
```

Comment:
s/It should have completed/Most installations complete/

We don't have a good way to judge how long an installation "should" take on any particular combination of network, equipment, and environment.

Reply:
When we say most, is it 51% or 99%?
What should the user do with such phrase?  Seems confusing